### ISSUE TITLE
Error of "Source line too long" when compiling HEMCO code with nvhpc/23.1 compiler

### ISSUE THREAD

**Issue by sjsprecious** [ESCOMP/CAM#871]
### What happened?

When I tried to build cam6_3_119 on Derecho with the nvhpc/23.1 compiler, the compilation failed with the error message below:
```
NVFORTRAN-S-0285-Source line too long (/glade/derecho/scratch/sunjian/CESM/components/cam/src/hemco/HEMCO/src/Core/hcoio_read_std_mod.F90: 65)
NVFORTRAN-S-0023-Syntax error - unbalanced parentheses (/glade/derecho/scratch/sunjian/CESM/components/cam/src/hemco/HEMCO/src/Core/hcoio_read_std_mod.F90: 65)
```
The culprit is that the `GC_72_EDGE_SIGMA` array in the `hcoio_read_std_mod.F90` code (indicated by the line number above) is too long (see a similar issue here: https://forums.developer.nvidia.com/t/compile-error-processing-long-comment-line/135899). The solution is to break the `GC_72_EDGE_SIGMA` array into multiple lines and use the continuation character.

### What are the steps to reproduce the bug?

- ./create_newcase --case /path_to_case --mach derecho --compiler nvhpc --mpilib mpich --compset F2000dev --res f19_f19_mg17 --walltime 01:00:00 --run-unsupported --queue main

- cd /path_to_case

- ./case.setup

- ./case.build

### What CAM tag were you using?

cam6_3_119

### What machine were you running CAM on?

Other (please explain below)

### What compiler were you using?

NVHPC

### Will you be addressing this bug yourself?

Any CAM SE can do this

### Extra info

I am building CAM on Derecho.

**Comment by jimmielin** [comment]
Hi @sjsprecious, thank you, this looks like it's an issue in the HEMCO upstream. I'll fix it over there and once it is tagged at HEMCO I can submit a pull request over here at CAM.

**Comment by yantosca** [comment]
https://github.com/geoschem/HEMCO/pull/234 has now been merged into the `dev/3.7.1` branch.  This will go into the next version of HEMCO.

**Comment by jimmielin** [comment]
Thanks @yantosca. Once HEMCO 3.7.1 is tagged I'll make a HEMCO_CESM external and a pull request here.

**Comment by cacraigucar** [comment]
This is tripping up more users as we are migrating to derecho and using the NVHPC compiler (since HEMCO is always compiled for any CAM run).  Could we perhaps get an updated HEMCO_CESM external soon?

**Comment by lizziel** [comment]
We are going to do a HEMCO 3.6.2 release containing the bug fix today. @jimmielin will update HEMCO_CESM accordingly asap.

**Comment by jimmielin** [comment]
This issue was resolved by PR #886. Once confirmed working please feel free to close the issue. Thanks!

**Comment by cacraigucar** [comment]
Closed in cam6_3_129
